### PR TITLE
Feature: Adding support for new workflow_job event.

### PR DESCRIPTION
**Pull request by npalm**
## Description
GitHub will add an event that will will be triggered on workflow jobs. With the new event the module can scale in a better way. Advantages could be:
- support multiple runner configurations based labels
- avoid scaling for other runners that als sents the check_run
- make the runners ephemera.

As first step the check_run event will be replace by the workflow_job event, the webhook will start checking if the runners labels are matching.  

## Design

For the moment we will support the check_run event for keeping backwards comptiable. New features will be designed for the new workflow_job event. As soon we are not able to keep supporting both events a new major relases will created for only the workflow_job event. 

### Lambda webhook
- Support for the new event will be added. 
- Lambda wil check if the workflow_job event runner labes match the extra labels that can be configured via the runner. This will avoid that the module will create runner instances that will never be used to schedule job.

### Lambda scale-up
- Support for the new event will be added

## Configuration
Consumers for the module can choose one (exclusive or) models to scale runners
1. The old way via check_run event. (app webhook, repo webhook, org webhook, enterprise webhook)
2. The new way via the workflow_job event. (app webhook, repo webhook, org webhook, enterprise webhook)

### CODE REVIEW

**Comment by npalm** [comment]
Tested with both events via org level runners

File: modules/runners/lambdas/runners/src/scale-runners/scale-up.test.ts [hunk 1]
```diff
@@ -26,15 +27,15 @@ jest.mock('./runners');
...
 const TEST_DATA: scaleUpModule.ActionRequestMessage = {
   id: 1,
-  eventType: 'check_run',
+  eventType: 'workflow_job',
   repositoryName: 'hello-world',
   repositoryOwner: 'Codertocat',
   installationId: 2,
 };
 
 const TEST_DATA_WITHOUT_INSTALL_ID: scaleUpModule.ActionRequestMessage = {
```
Comment (bartgolsteijn) [inline modules/runners/lambdas/runners/src/scale-runners/scale-up.test.ts:36]: Consider renaming to `TEST_DATA_WITH_ZERO_INSTALL_ID` or remove field

File: modules/runners/lambdas/runners/src/scale-runners/scale-up.test.ts [hunk 2]
```diff
@@ -350,16 +364,16 @@ describe('scaleUp with public GH', () => {
     beforeEach(() => {
       process.env.ENABLE_ORGANIZATION_RUNNERS = 'true';
       expectedRunnerParams = { ...EXPECTED_RUNNER_PARAMS };
-      expectedRunnerParams.runnerServiceConfig =
-        `--url https://github.com/${TEST_DATA.repositoryOwner} --token 1234abcd `;
+      // eslint-disable-next-line max-len
+      expectedRunnerParams.runnerServiceConfig = `--url https://github.com/${TEST_DATA.repositoryOwner} --token 1234abcd `;
```
Comment (bartgolsteijn) [inline modules/runners/lambdas/runners/src/scale-runners/scale-up.test.ts:368]: Consider splitting over two lines (after `=`) and removing eslint warning suppression

File: modules/runners/lambdas/runners/src/scale-runners/scale-up.test.ts [hunk 3]
```diff
@@ -407,9 +426,8 @@ describe('scaleUp with public GH', () => {
       expectedRunnerParams = { ...EXPECTED_RUNNER_PARAMS };
       expectedRunnerParams.runnerType = 'Repo';
       expectedRunnerParams.runnerOwner = `${TEST_DATA.repositoryOwner}/${TEST_DATA.repositoryName}`;
-      expectedRunnerParams.runnerServiceConfig = `--url ` +
-        `https://github.com/${TEST_DATA.repositoryOwner}/${TEST_DATA.repositoryName} ` +
-        `--token 1234abcd `;
+      expectedRunnerParams.runnerServiceConfig =
+        `--url ` + `https://github.com/${TEST_DATA.repositoryOwner}/${TEST_DATA.repositoryName} ` + `--token 1234abcd `;
```
Comment (bartgolsteijn) [inline modules/runners/lambdas/runners/src/scale-runners/scale-up.test.ts:430]: Consider replacing string addition by a string interpolation.

**Comment by npalm** [comment]
Closes #726

**Comment by mcaulifn** [comment]
Consider holding this one till #941 has been merged?

**Comment by npalm** [comment]
> Consider holding this one till #941 has been merged?

@mcaulifn in case we merge this first, I will help you with the merge conflicts :)

**Review by bartgolsteijn — DISMISSED** [review]
lgtm

**Comment by npalm** [comment]
Rebased with develop, @bartgolsteijn @gertjanmaas please can one of check and approve.

File: README.md [hunk 4]
```diff
@@ -36,11 +39,16 @@ A logical question would be why not Kubernetes? In the current approach we stay
 
 ## Overview
 
-The moment a GitHub action workflow requiring a `self-hosted` runner is triggered, GitHub will try to find a runner which can execute the workload. This module reacts to GitHub's [`check_run` event](https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#check_run) for the triggered workflow and creates a new runner if necessary.
+The moment a GitHub action workflow requiring a `self-hosted` runner is triggered, GitHub will try to find a runner which can execute the workload. This module reacts to GitHub's [`check_run event](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#check_run) or [`workflow_job` event](https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#workflow_job) for the triggered workflow and creates a new runner if necessary.
```
Comment (bartgolsteijn) [inline README.md:42]:
```
[`check_run event] -> [`check_run` event]
```

**Review by bartgolsteijn — DISMISSED** [review]

**Review by bartgolsteijn — APPROVED** [review]
LGTM

**Review by bartgolsteijn — APPROVED** [review]

**Pull request merged**